rllib/agents/trainer.py

```diff
@@ -1295,6 +1301,12 @@ def training_iteration(self) -> ResultDict:
         else:
             train_results = multi_gpu_train_one_step(self, train_batch)
 
+        # Update weights - after learning on the local worker - on all remote
+        # workers.
+        if self.workers.remote_workers():
```

Comment:
do we always sync weights after each iteration?
for off policy methods, how often weights get synced feels like a hyper-param?

Reply:
Remember, this is only the basic, default iteration function. Off-policy algos will define a different one.

Reply:
Just wanted to move this out of the `train_one_step` function, as I felt weight synching is kind of it's own logical unit within an iteration.